### PR TITLE
fix: next_alarm sensor not showing proper snooze state

### LINKED ISSUE #3402: [Bug]: next_alarm sensor is not showing proper snooze state for active alarms
### Describe the bug

When checking sorted_active alarms under the next_alarm sensor, it is not indicating the proper Snoozed state of the alarm (in particular, I'm trying to find snoozed time).  For _reoccurring_ alarms, the status is always shown as 'ON' instead of 'SNOOZED'. And, for all alarms, _single time or reoccurring_, snoozedToTime: is always null

### To Reproduce

Define a **reoccurring** alarm on Alexa device, when the alarm sounds, snooze it.
Check the next_alarm sensor in Dev Tools and check attributes for status: (will always show 'ON') and snoozedToTime: (will always show null)

Define a **single non-reoccurring** alarm on Alexa device, when the alarm sounds, snooze it.
Check the next_alarm sensor in Dev Tools and check attributes for status: (will correctly show 'SNOOZED') but snoozedToTime: will still always show null

### Expected behavior

I expect that status: would be 'SNOOZED' but shows 'ON' and snoozedToTime: to be the datetime of the snoozed alarm but shows as null.

The attached file shows sorted_active attribute of 

[next_alarm sorted_active test.txt](https://github.com/user-attachments/files/25625594/next_alarm.sorted_active.test.txt)

 where the device has 12:05 pm reoccurring alarm and 12:10 pm single alarm both snoozed.  

### Screenshots

_No response_

### Home Assistant Version

2026.2.3

### Alexa Media Player Version

5.14.0

### Alexa Media Player API library Version

1.29.17

### Have you entered a 52-character key for Amazon 2SV via Authenticator App login?

Yes

### Is the HA core integration Alexa Devices installed?

No

### Amazon Domain

amazon.com (US)

### Amazon Domain (if Other)

_No response_

### Debug Logs (alexa_media & alexapy)

```text

```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by danielbrunt57**
Add datetime coercion and alarm snooze normalization.

Fixes #3402 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of snoozed Alexa alarms: datetimes from epochs, strings, or naive values are normalized to the correct timezone; snoozed alarms are consistently labeled, ordered by the snooze timestamp, and preserved during recurring updates. Active-selection now correctly includes/excludes snoozed and expired alarms and exposes snooze timing in debug output.

* **Tests**
  * Added tests for snooze normalization, active-selection rules, datetime parsing, and boundary cases.

* **Chores**
  * Introduced a threshold constant to distinguish epoch seconds vs milliseconds.

### CODE REVIEW

File: custom_components/alexa_media/sensor.py [hunk 1]
```diff
@@ -601,13 +601,70 @@ def __init__(
...
+            next_item["snoozedToTime"] = snoozed_to
+            next_item[self._sensor_property] = snoozed_to
+            return value
+
+        if status == "SNOOZED" and snoozed_to is None and alarm_when is not None:
+            next_item["snoozedToTime"] = alarm_when
+            next_item[self._sensor_property] = alarm_when
+            return value
+
+        return value
```
[the remark this answers is not in the document]
Comment (danielbrunt57) [reply custom_components/alexa_media/sensor.py:659]: Refactored to address comments.

**Comment by danielbrunt57** [comment]
@coderabbitai Please draft pytest cases covering the four key states (ON, SNOOZED with future snoozedToTime, SNOOZED with missing snoozedToTime, expired SNOOZED)

**Review by alandtse — APPROVED** [review]
